File: go/gin/internal/modules/transaction/module.go
package transaction

import (
	"tcc/internal/modules/account"
	auth "tcc/internal/modules/auth/middleware"

	"github.com/gin-gonic/gin"
)

func TransactionModule(router *gin.Engine) {
	service := TransactionService{
		repository: struct{account account.AccountRepository; transaction TransactionsRepository}{
			account: account.AccountRepository{},
			transaction: TransactionsRepository{},
		},
	}
	controller := TransactionController{
		service: service,
	}

	group := router.Group("/transacoes")
	{
		group.GET("", auth.AuthMiddleware(), controller.FindAll)
		group.POST("", auth.AuthMiddleware(), controller.Create)
	}
}

File: rust/actix/src/modules/transaction/module.rs
use actix_web::{web, Scope, dev::ServiceFactory};
use actix_web::dev::ServiceRequest;
use actix_web::dev::ServiceResponse;
use actix_web::Error;
use crate::modules::auth::middleware::AuthMiddleware;
use crate::modules::transaction::controller::TransactionController;

pub fn transaction_module() -> Scope<impl ServiceFactory<ServiceRequest, Config = (), Response = ServiceResponse, Error = Error, InitError = ()>> {
    web::scope("/transacoes")
        .wrap(AuthMiddleware)
        .service(
            web::resource("")
                .route(web::get().to(TransactionController::find_all))
                .route(web::post().to(TransactionController::create))
        )
} 